packages/rebber-plugins/src/preprocessors/codeVisitor.js

```diff
@@ -1,41 +1,53 @@
 const visit = require('unist-util-visit')
+const appendix = require('../type/appendix')
 
 module.exports = (ctx, tree) => {
-  const title = ctx.codeAppendiceTitle || 'Appendices'
+  ctx.overrides.appendix = appendix
   return (node) => {
-    const inAppendix = []
-    let appendixIndex = 1
-
+    let appendix = tree.children[tree.children.length - 1]
+    if (!appendix || appendix.type !== 'appendix') {
+      appendix = {
+        type: 'appendix',
+        children: [],
+      }
+      tree.children.push(appendix)
+    }
+    let appendixIndex = appendix.children.length + 1
     visit(node, 'code', (innerNode, index, _parent) => {
-      inAppendix.push({
+      appendix.children.push({
         type: 'paragraph',
         children: [
           {
-            type: 'definition',
-            identifier: `appendix-${appendixIndex}`,
-            referenceType: 'full',
-            children: [{type: 'text', value: 'code'}],
+            type: 'heading',
+            children: [
+              {
+                type: 'definition',
+                identifier: `appendix-${appendixIndex}`,
+                url: `${ctx.codeAppendiceTitle || 'Appendix'} ${appendixIndex}`,
+                referenceType: 'full',
+                children: [{type: 'text', value: 'code'}],
```

Comment:
I didn't noticed this children at the first time, why is it here, I don't think it is needed. It's not even parsed it seems...

Reply:
Oh yes, this is the rest of the old code

Reply:
removed